### PR TITLE
Stayview and removal of graphql

### PR DESCRIPTION
**Pull request by jonathan-s**
This took way longer than I expected. This PR removes the usage of graphql from the StayView. This is thus the first step in removing graphql completely from modernomad. 

In this PR react-router is also upgraded to the latest version.

### CODE REVIEW

File: client/js/components/booking/RoomIndexOrDetail.jsx [hunk 1]
```diff
@@ -1,140 +1,72 @@
...
+      arrive: filters.dates.arrive.format('MM/DD/YYYY'),
+      depart: filters.dates.depart.format('MM/DD/YYYY')
     }
-
-    browserHistory.push({
-      pathname: path,
-      query: formattedDates
+    var path = `/locations/${this.location_name}/stay/`
+    let jsonUrl = `/locations/${this.location_name}/json/room`
+    axios.get(jsonUrl, {params: formattedDates})
```
Comment (bfirsh) [inline client/js/components/booking/RoomIndexOrDetail.jsx]: How does this handle errors?

File: modernomad/settings/common.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
 # Local time zone for this installation. Choices can be found here:
 # http://en.wikipedia.org/wiki/List_of_tz_zones_by_name
 TIME_ZONE = 'America/Los_Angeles'
+DATE_FORMAT = '%m/%d/%Y'
```
Comment (bfirsh) [inline modernomad/settings/common.py]: Maybe this is opportune time to switch to a not terrible date format. ;)

I vote for ISO `%Y-%m-%d`

**Comment by jonathan-s** [comment]
@bfirsh Made the requested changes :)

**Comment by bfirsh** [comment]
Cypress tests are still failing for some reason... restarted the build...

**Comment by bfirsh** [comment]
I restarted the build so I think these failures are real. :(

**Comment by bfirsh** [comment]
Looks like there was a missing file, which I have created. Let's see if this works...

Irritatingly webpack should exit with an error as part of the build process about this, but we're using such an old version of webpack that [I don't think it does that](https://github.com/webpack-contrib/webpack-command/pull/44).

**Comment by bfirsh** [comment]
Also, the "Use a different querystring package" commit completely rewrote package-lock.json with a different format.

@jonathan-s - I think you might be running an old version of npm. If you're using Docker, it is possible to install npm packages inside the Docker container, but we should probably document or script it because it isn't trivial. Something like `docker-compose run web sh -c "cd client && npm install [package]"`.

**Comment by bfirsh** [comment]
Woo :tada:

**Comment by jonathan-s** [comment]
@bfirsh Thanks a lot for doing those finishing touches!

**Pull request merged**